uint64 inc
uint8 on
uint8 horizontal